gilmour.go

```diff
@@ -475,3 +510,29 @@ func (g *Gilmour) publish(topic string, msg *Message) error {
 
 	return g.backend.Publish(topic, msg)
 }
+
+func isNetworkError(err error) bool {
+	if _, ok := err.(net.Error); ok {
+		return true
+	} else {
+		return false
+	}
+}
+
+func isExpired(timeout time.Duration, startTime time.Time) bool {
+	return time.Now().After(startTime.Add(timeout))
+}
+
+func try(r RetryConf, fn func() error) (err error) {
+	startTime := time.Now()
+	for {
+		err = fn()
+		if isExpired(r.Timeout, startTime) || !isNetworkError(err) || err == nil {
```

Comment:
@meson10 Yes, in case of network error, it always raise net.Error.
So basically checking expiry is a way to check if the request has timed out or not.